frontend/src/metabase/core/components/FormCategoryInput/FormCategoryInput.tsx

```diff
@@ -0,0 +1,67 @@
+import React, {
+  ChangeEvent,
+  forwardRef,
+  ReactNode,
+  Ref,
+  useCallback,
+} from "react";
+import { useField } from "formik";
+import { useUniqueId } from "metabase/hooks/use-unique-id";
+import FormField from "metabase/core/components/FormField";
+
+import type Field from "metabase-lib/metadata/Field";
```

Comment:
This component shouldn't be in `core/components` because it uses metabase-lib entities. I think we could either move it to `metabase/components` (like we do now in these cases) or create another directory for such components like discussed here https://www.notion.so/metabase/42-Structure-top-level-directories-c64f3234ae334b3dbb591e118f415fab

Reply:
that makes sense. I wonder if we could put a readme.md in core/components outlining what should and should not go in the file? I think that will be helpfpul down the road.

Reply:
@ranquild moved it, mind taking another look at the PR?